### PR TITLE
Add pipefunc

### PR DESCRIPTION
**Pull request by basnijholt**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by basnijholt** [comment]
@conda-forge/help-python can someone please review and merge this?

File: recipes/pipefunc/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/pipefunc-{{ version }}.tar.gz
+  sha256: 988ef01177d73cfd8f38c86999ac4819818c562e75aee101b34fc6502877b623
+
+build:
+  skip: true  # [py<38]
```
Comment (ocefpaf) [inline recipes/pipefunc/meta.yaml:13]: I believe this one can be noarch.

```suggestion
  noarch: python
```

File: recipes/pipefunc/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    - setuptools
+    - wheel
+    - versioningit
+    - pip
+  run:
+    - python
+    - networkx
+    - psutil
+    - versioningit
+    - typing-extensions  # [py<=39]
```
Comment (ocefpaf) [inline recipes/pipefunc/meta.yaml:29]: This works as a placeholder py>39.

```suggestion
    - typing-extensions
```

File: recipes/pipefunc/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+requirements:
+  host:
+    - python
+    - setuptools
+    - wheel
+    - versioningit
+    - pip
+  run:
+    - python
```
Comment (ocefpaf) [inline recipes/pipefunc/meta.yaml:25]:
```suggestion
    - python >=3.8
    - setuptools
    - wheel
    - versioningit
    - pip
  run:
    - python >=3.8
```

**Comment by basnijholt** [comment]
Thanks a lot for your review @ocefpaf! I have applied your suggestions.

File: recipes/pipefunc/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/pipefunc-{{ version }}.tar.gz
+  sha256: 988ef01177d73cfd8f38c86999ac4819818c562e75aee101b34fc6502877b623
+
+build:
+  noarch: python
```
Comment (ocefpaf) [inline recipes/pipefunc/meta.yaml:13]: This should fix Windows:

```suggestion
  noarch: python
  entry_points:
    - pipefunc-launcher = pipefunc._server_support.launcher:main
```

Comment (basnijholt) [reply recipes/pipefunc/meta.yaml:13]: Thanks for noticing this. This script was actually removed before v0.1.0. I just released v0.2.0 and fixed this.

**Pull request merged**